src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -172,7 +172,7 @@ public function withCount($relations)
 
             unset($alias);
 
-            if (count($segments) == 3 && Str::lower($segments[1]) == 'as') {
+            if (count($segments) == 3 && Str::lower($segments[1]) === 'as') {
```

Comment:
This PR is only focusing on stricter comparisons for string, not for int and float
Or else this PR would be be probably to big to review